File: Raffle_Name_Picker.py
import random

def raffle_name_picker():
    participants = []
    while True:
        print("1. Add participant")
        print("2. Pick a winner")
        print("3. Exit")
        choice = input("Enter your choice: ")
        if choice == '1':
            name = input("Enter participant's name: ")
            participants.append(name)
        elif choice == '2':
            if len(participants) == 0:
                print("No participants added yet.")
            else:
                winner = random.choice(participants)
                print("The winner is: " + winner)
        elif choice == '3':
            break
        else:
            print("Invalid choice. Please choose again.")

if __name__ == "__main__":
    raffle_name_picker()

File: Raffle_Name_Picker.java
```java
import java.util.ArrayList;
import java.util.Random;
import java.util.Scanner;

public class RaffleNamePicker {
    public static void main(String[] args) {
        ArrayList<String> names = new ArrayList<>();
        Scanner scanner = new Scanner(System.in);
        String input = "";

        while (!input.equals("end")) {
            System.out.println("Enter a name or type 'end' to finish:");
            input = scanner.nextLine();
            if (!input.equals("end")) {
                names.add(input);
            }
        }

        Random random = new Random();
        String winner = names.get(random.nextInt(names.size()));
        System.out.println("The winner is: " + winner);
    }
}
```